src/token/src/canister/dip20_transactions.rs

```diff
@@ -686,4 +686,43 @@ mod tests {
         let canister = test_canister();
         canister.getTransaction(Nat::from(2));
     }
+
+    #[test]
+    fn get_user_transactions() {
+        let canister = test_canister();
+        const COUNT: usize = 5;
+        for _ in 1..COUNT {
+            canister.transfer(bob(), Nat::from(10), None).unwrap();
+        }
+        canister.transfer(john(), Nat::from(10), None).unwrap();
+        canister.transfer(xtc(), Nat::from(10), None).unwrap();
+        let txs = canister.getUserTransactions(alice(), Nat::from(0), Nat::from(10));
+
+        assert_eq!(txs.len(), 7);
+        assert_eq!(txs[0].to, xtc());
+
+        let txs = canister.getUserTransactions(alice(), Nat::from(0), Nat::from(2));
```

Comment:
add an essertion to check the order of transactions is correctly reversed, thanks